### DISCUSSION TITLE
How to get Signed Measurement in SPDM 1.2.0

### DISCUSSION THREAD

**Discussion by Ankur-80** [DMTF/libspdm#2889] [Q&A] [answered]
As per Redfish DMTF Spec, Signed Measurement is Concatenate of VCA and Measurement Request & Response if Supported version is 1.2. Which is basically L1L2 from my understanding. Is there any API or anything part of requester context in libspdm to get the this data directly?

**L1/L2 = Concatenate(`VCA`, GET_MEASUREMENTS_REQUEST1, MEASUREMENTS_RESPONSE1, ...,
GET_MEASUREMENTS_REQUESTn-1, MEASUREMENTS_RESPONSEn-1,
GET_MEASUREMENTS_REQUESTn, MEASUREMENTS_RESPONSEn)**

As per Redfish DMTF Spec,
                **"SignedMeasurements": {
                    "description": "Base64-encoded cryptographic signed statement generated by the signer.",
                    "longDescription": "This property shall contain the cryptographic signed statement over the given nonce and measurement blocks corresponding to the requested measurement indices.  If the SPDM version is 1.2, this value shall be a concatenation of SPDM 'VCA' and 'GET_MEASUREMENTS' requests and responses exchanged between the SPDM Requester and the SPDM Responder.  If SPDM version is 1.0 or 1.1, this value shall be a concatenation of SPDM 'GET_MEASUREMENTS' requests and responses exchanged between the SPDM Requester and the SPDM Responder.  The last 'MEASUREMENTS' response shall contain a signature generated over the 'L2' string by the SPDM Responder.",
                    "readonly": true,
                    "type": "string"
                },**

**Comment by steven-bellock** [comment, marked as answer]
See https://github.com/DMTF/libspdm/blob/main/doc/user_guide.md#message-logging.
